mithril-aggregator/src/commands/genesis_command.rs

```diff
@@ -1,18 +1,130 @@
+use std::{path::PathBuf, sync::Arc};
+
 use anyhow::Context;
 use clap::{Parser, Subcommand};
-use config::{builder::DefaultState, ConfigBuilder};
+use config::{builder::DefaultState, ConfigBuilder, Map, Value};
+use serde::{Deserialize, Serialize};
+use slog::{debug, Logger};
+
 use mithril_common::{
+    chain_observer::ChainObserverType,
     crypto_helper::{ProtocolGenesisSecretKey, ProtocolGenesisSigner},
-    entities::HexEncodedGenesisSecretKey,
+    entities::{
+        CardanoTransactionsSigningConfig, HexEncodedGenesisSecretKey,
+        HexEncodedGenesisVerificationKey, ProtocolParameters,
+    },
     StdResult,
 };
-use slog::{debug, Logger};
-use std::{path::PathBuf, sync::Arc};
+use mithril_doc::{Documenter, StructDoc};
 
 use crate::{
-    dependency_injection::DependenciesBuilder, tools::GenesisTools, ServeCommandConfiguration,
+    dependency_injection::DependenciesBuilder, tools::GenesisTools, ConfigurationSource,
+    ExecutionEnvironment,
 };
 
+#[derive(Debug, Clone, Serialize, Deserialize, Documenter)]
+pub struct GenesisCommandConfiguration {
+    /// Cardano CLI tool path
+    #[example = "`cardano-cli`"]
+    pub cardano_cli_path: PathBuf,
+
+    /// Path of the socket used by the Cardano CLI tool
+    /// to communicate with the Cardano node
+    #[example = "`/tmp/cardano.sock`"]
+    pub cardano_node_socket_path: PathBuf,
+
+    /// Cardano Network Magic number
+    ///
+    /// useful for TestNet & DevNet
+    #[example = "`1097911063` or `42`"]
+    pub network_magic: Option<u64>,
+
+    /// Cardano network
+    #[example = "`testnet` or `mainnet` or `devnet`"]
+    network: String,
+
+    /// Cardano chain observer type
+    pub chain_observer_type: ChainObserverType,
+
+    /// Directory of the Cardano node store.
+    pub db_directory: PathBuf,
+
+    /// Directory to store aggregator data (Certificates, Snapshots, Protocol Parameters, ...)
+    #[example = "`./mithril-aggregator/stores`"]
+    pub data_stores_directory: PathBuf,
+
+    /// Genesis verification key
+    pub genesis_verification_key: HexEncodedGenesisVerificationKey,
```

Comment:
This information is probably needed only when signing or bootstrapping

Reply:
This probably means that multiple configurations are needed for the sub commands